src/test/java/seedu/address/logic/commands/DeleteStudentCommandTest.java

```diff
@@ -5,58 +5,152 @@
+    // Ensure deletion by provided name and provided student number works when student with correct details exists
+    @Test
+    public void execute_singleStudentDeleteByNameAndStudentNumber_success() throws Exception {
+        Student validStudent = new StudentBuilder().withName("John Ng").withStudentNumber("A1234567X").build();
+        DeleteStudentCommandTest.ModelStubWithStudent modelStub =
+                new DeleteStudentCommandTest.ModelStubWithStudent(validStudent);
+
+        DeleteStudentCommand command = new DeleteStudentCommand(new Name("John Ng"), new StudentNumber("A1234567X"));
+
+        CommandResult result = command.execute(modelStub);
+
+        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent.getName()
+                        + " " + validStudent.getStudentNumber()),
+                result.getFeedbackToUser());
+    }
+
+    // Ensure deletion by provided name and provided student number works with duplicate student names
+    @Test
+    public void execute_duplicateStudentDeleteByNameAndStudentNumber_success() throws Exception {
+        Student validStudent1 = new StudentBuilder().withName("John Ng").withStudentNumber("A1234567X").build();
+        Student validStudent2 = new StudentBuilder().withName("John Ng").withStudentNumber("A0123456Y").build();
+        Student validStudent3 = new StudentBuilder().withName("John Ng").withStudentNumber("A9876543Z").build();
+        Student validStudent4 = new StudentBuilder().withName("John Ng").withStudentNumber("A1111111B").build();
+        DeleteStudentCommandTest.ModelStubWithStudent modelStub =
+                new DeleteStudentCommandTest.ModelStubWithStudent(validStudent1, validStudent2,
+                        validStudent3, validStudent4);
+
+        DeleteStudentCommand firstCommand = new DeleteStudentCommand(new Name("John Ng"),
+                new StudentNumber("A1234567X"));
+
+        CommandResult firstResult = firstCommand.execute(modelStub);
+
+        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent1.getName()
+                        + " " + validStudent1.getStudentNumber()),
+                firstResult.getFeedbackToUser());
+
+        assertEquals(FXCollections.observableArrayList(validStudent2, validStudent3, validStudent4),
+                modelStub.getFilteredStudentList());
 
+        DeleteStudentCommand secondCommand = new DeleteStudentCommand(new Name("John Ng"),
+                new StudentNumber("A9876543Z"));
+
+        CommandResult secondResult = secondCommand.execute(modelStub);
+
+        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent3.getName()
+                        + " " + validStudent3.getStudentNumber()),
+                secondResult.getFeedbackToUser());
+
+        assertEquals(FXCollections.observableArrayList(validStudent2, validStudent4),
+                modelStub.getFilteredStudentList());
```

Comment:
This might be a little redundant since it was tested for in the previous few lines.